app/models/rad_consumer_session.rb

```diff
@@ -29,4 +31,22 @@ def store(firm_result, profile_url, most_recent_search_url)
   def firm_already_present?(firm_result)
     firms.any? { |firm_hash| firm_result.id == firm_hash['id'] }
   end
+
+  def most_recent_search_path_values(params)
+    {
+      'en' => search_path_for(params, 'en'),
+      'cy' => search_path_for(params, 'cy')
+    }
+  end
+
+  def search_path_for(params, locale)
+    search_path(search_form: params[:search_form], locale: locale)
+  end
+
+  def profile_path_values(params)
```

Comment:
And similarly a rename to `locale_to_profile_path_mappings` would make this easier to understand to me.